ocaml/typing/jkind_intf.ml

```diff
@@ -33,8 +33,14 @@ module type Sort = sig
   (** A sort variable that can be unified during type-checking. *)
   type var
 
+  type const =
+    | Const_base of base
```

Comment:
Aha - we can.  We had name conflicts because this type was exposed in the outer module, for reasons Liam explains [here](https://github.com/ocaml-flambda/flambda-backend/pull/2676#discussion_r1646730077), but since Sort.t no longer depends on const after this PR, we can unwind that and delete the `const` type, keeping only the `Const.t` version.  I've done all that.